pr_insight/git_providers/github_provider.py

```
@@ -475,8 +476,17 @@ def publish_code_suggestions(self, code_suggestions: list) -> bool:
-        comment.edit(body=body)
+        try:
+            body = self.limit_output_characters(body, self.max_comment_chars)
+            comment.edit(body=body)
+        except GithubException as e:
+            if hasattr(e, "status") and e.status == 403:
+                # Log as warning for permission-related issues (usually due to polling)
+                get_logger().warning(
+                    "Failed to edit github comment due to permission restrictions",
+                    artifact={"error": e})
+            else:
+                get_logger().exception(f"Failed to edit github comment", artifact={"error": e})
```

Ruff (0.8.2)

f-string without any placeholders

(F541)